Add hpe-design-system-codemods (#5326)
grommet/hpe-design-system: packages/codemods/README.md

> # Grommet Theme HPE Codemod

Reviewer: ```suggestion
# HPE Design System Codemods
```

> A codemod project for migrating Grommet theme versions.

Reviewer: ```suggestion
A codemod project to assist migrations for package versions related to the HPE Design System.

Available codemod transforms:
  - [Grommet-theme-hpe v6 to v7](#Grommet-theme-hpe-v6-to-v7)
```

## Features

- Handles JS, JSX, TS, and TSX files
- Dry run and verbose options

## Usage

### 1. Install dependencies

```
yarn install
```

### 2. Run the codemod

you can run the codemod directly with npx:

```

> npx hpe-design-system-codemods migrate-theme-v6-to-v7 <path> [options]

Reviewer: ```suggestion
npx hpe-design-system-codemods <codemode transform> <path> [options]
```

Author: Fixed

```

## Codemod transforms

### Grommet-theme-hpe v6 to v7

- Automated migration of t-shirt size props (e.g., `gap`, `margin`, `pad`, `thickness`, `border`, `height`, `width`, `round`, etc.)
- Scan mode to detect t-shirt sizes without making changes.

> #### Options

Reviewer: Insert this usage, specific to this particular codemod, ahead of Options?

#### Usage

```
npx hpe-design-system-codemods migrate-theme-v6-to-v7 <path> [options]
```

#### Options
